### PR TITLE
update `teleport-kube-agent` reference + generate it

### PR DESCRIPTION
**Pull request by hugoShaka**
This PR generates the `teleport-kube-agent` reference and updates most of the chart documentation to fix:

- missing values
- remove deprecated values
- out-of-date or incorrect information

## Note for reviewers:

Most of the documentation has been re-written; this is not a simple copy-paste.

As discussed with @ptgott , it's better to merge this PR and fix wording and syntax _a posteriori_ rather than spending another month reviewing this.

### CODE REVIEW

**Review by marcoandredinis — APPROVED** [review]

File: docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx [hunk 1]
```diff
@@ -0,0 +1,1627 @@
+
+## `roles`
+
+| Type | Default |
+|------|---------|
+| `string` | `"kube"` |
+
+`roles` is a comma-separated list of services which should be enabled
```
Comment (ptgott) [inline docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx:8]: "should be enabled" sounds like it's nondeterministic—I'd recommend "will be enabled".

File: docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx [hunk 2]
```diff
@@ -0,0 +1,1627 @@
...
+when running the `teleport-kube-agent` chart.
+
+| Services                     | Value for `roles` | Mandatory additional settings for this role |
+|------------------------------|-------------------|---------------------------------------------|
+| Teleport Kubernetes service  | `kube`            | [`kubeClusterName`](#kubeclustername)       |
+| Teleport Application service | `app`             | [`apps`](#apps) or [`appResources`](#appresources) |
+| Teleport Database service    | `db`              | [`databases`](#databases) or [`databaseResources`](#databaseresources) |
+| Teleport Discovery service   | `discovery`       | [`kubeClusterName`](#kubeclustername) |
+
+For example: `roles: kube,app,discovery`.
```
Comment (ptgott) [inline docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx:18]: I would make this a YAML code snippet so it's more immediately clear what this means.

File: docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx [hunk 3]
```diff
@@ -0,0 +1,1627 @@
...
+
+## `proxyAddr`
+
+| Type | Default |
+|------|---------|
+| `string` | `""` |
+
+`proxyAddr` provides the public-facing Teleport proxy endpoint which should be used to join the cluster. This is the same URL that is used
```
Comment (ptgott) [inline docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx:26]:
```suggestion
`proxyAddr` provides the public-facing Teleport Proxy Service endpoint which should be used to join the cluster. This is the same URL that is used
```

File: docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx [hunk 4]
```diff
@@ -0,0 +1,1627 @@
...
+to access the web UI of your Teleport cluster. It is the same as the value configured for `proxy_service.public_addr` in a traditional
```
Comment (ptgott) [inline docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx:27]: Teleport Cloud users won't necessarily think of `proxy_service.public_addr` as traditional, so I'd find another way to put this.

File: docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx [hunk 5]
```diff
@@ -0,0 +1,1627 @@
...
+Teleport cluster. The port used is usually either 3080 or 443.
+
+Here are a few examples:
+
+| Deployment method             | Example `proxy_service.public_addr` |
+|-------------------------------|-------------------------------------|
+| On-prem Teleport cluster      | `teleport.example.com:3080`         |
+| Teleport Cloud cluster        | `example.teleport.sh:443`           |
+| `teleport-cluster` Helm chart | `teleport.example.com:443`          |
```
Comment (ptgott) [inline docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx:36]: I'm not sure we need this table. If users aren't sure which port the Proxy Service listens on, maybe we should add a `curl <proxy_addr>/webapi/ping` example?

File: docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx [hunk 6]
```diff
@@ -0,0 +1,1627 @@
...
+## `enterprise`
+
+| Type | Default |
+|------|---------|
+| `bool` | `false` |
+
+`enterprise` controls if the `teleport-kube-agent` chart should deploy
+the OSS version or the enterprise version of the container image.
+This should be set to `true` for connecting to Teleport Cloud and self-hosted
+Teleport Enterprise clusters to use enterprise features within the Kube Agent.
```
Comment (ptgott) [inline docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx:47]: Does "Kube Agent" refer to the agent as a whole (regardless of services it runs) or just the Kubernetes Service?

File: docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx [hunk 7]
```diff
@@ -0,0 +1,1627 @@
...
+Its usage should be preferred.
+
+### `joinParams.method`
+
+| Type | Default |
+|------|---------|
+| `string` | `"token"` |
+
+`joinParams.method` controls which join method will be used by the
+instance join the Teleport cluster.
```
Comment (ptgott) [inline docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx:88]: "instance to join"

File: docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx [hunk 8]
```diff
@@ -0,0 +1,1627 @@
...
+
+See [the join method reference](../../join-methods.mdx) for the list of possible
+values, the implications of each join method, and guides to setup each method.
```
Comment (ptgott) [inline docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx:91]: "Setup" is a noun. "Set up" is a verb phrase.

File: docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx [hunk 9]
```diff
@@ -0,0 +1,1627 @@
...
+
+`apps` is a static list of applications that should be proxied by
+the agent. See [the Teleport Application access documentation](../../../application-access/introduction.mdx)
+for more details.
+
+Proxied applications can be defined statically (through this value) or dynamically
+(through the [`appResources`](#appResources) value).
+One of `apps` and `appResources` is required if the chart `roles` contains `app`.
+
+You can specify multiple apps by adding additional list elements.
```
Comment (ptgott) [inline docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx:147]: "adding additional list elements" -> "adding elements to the list" (no need to repeat "add")

File: docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx [hunk 10]
```diff
@@ -0,0 +1,1627 @@
...
+    labels:
+      purpose: monitoring
+  - name: jenkins
+    uri: http://jenkins:8080
+    labels:
+      purpose: ci
+```
+
+<Admonition type="tip" title="Supported values">
+  You can see a list of all the supported[values which can be used in a Teleport
```
Comment (ptgott) [inline docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx:163]: -> "You can see a list of all the supported values that can be used in a Teleport Application Service configuration in the [Application Service Configuration Reference]"

- Describes the link destination
- The config is for the Application Service

File: docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx [hunk 11]
```diff
@@ -0,0 +1,1627 @@
...
+
+## `appResources`
+
+| Type | Default |
+|------|---------|
+| `list` | `[]` |
+
+`appResources` is a set of labels the agent will monitor. Any application
+matching those labels will be proxied by the agent. See [the Teleport
+Application access documentation](../../../application-access/introduction.mdx)
```
Comment (ptgott) [inline docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx:175]: Should be "application access" or "Application Service"

File: docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx [hunk 12]
```diff
@@ -0,0 +1,1627 @@
...
+appResources:
+  - labels:
+      "env": "prod"
+  - labels:
+      "env": "test"
+```
+
+<Admonition type="tip" title="Example">
+Once `appResources` is set, you can dynamically register application with
+`tsh` by following [this guide](../../../application-access/guides/dynamic-registration.mdx).
```
Comment (ptgott) [inline docs/pages/reference/helm-reference/includes/zz_generated.teleport-kube-agent.mdx:194]: "this guide" -> "Dynamic App Registration"

**Review by ptgott — COMMENTED** [review]
I've left a partial review. I need to resume at `awsDatabases`.

I'm going to do my best to keep my comments are sparse as possible for the remainder of the review. I know I've picked up on some minor style issues, and there are likely to be more of these, but I also don't want a style review to block this PR unnecessarily.

**Review by webvictim — APPROVED** [review]

**Review by ptgott — APPROVED** [review]
I'm going to go ahead and approve this without reviewing much of it. My reasoning is that the `teleport-kube-agent` values file has no values without a comment, which means that no values are missing from the reference, since the generator is based on extracting comments from the default values file. As a result, the reference is technically complete. 

Any remaining issues would result from:
- The style/correctness of the values file comments the reference guide is based on
- Small issues with the generator that don't affect the ability to extract fields from the values file

I think we can forgive these for now since otherwise, this PR will take an undue amount of time to review. Since the generator is, in my mind, here to stay, we can iterate on it to improve this and other Helm reference pages.

**Review by ptgott — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Pull request merged**